### PR TITLE
Add pep8-naming to pre-commit hooks and fixes incorrect naming conventions

### PR DESCRIPTION
**Pull request by CaedenPH**
### Describe your change:
Fixes naming conventions and adds `pep-8` naming to `pre-commit` hooks
#7038 

This code **does not** edit any variable names other than changing from `mixedCase` to `snake_case`.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by CaedenPH** [comment]
Will work on failing checks tommorow

**Comment by cclauss** [comment]
# WOW!!!
You win the gold star if you can land this one!!  I had no idea it would 130+ files!!

File: arithmetic_analysis/lu_decomposition.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from __future__ import annotations
 
 import numpy as np
-import numpy.typing as NDArray
+import numpy.typing as NDArray  # noqa: N812
```
Comment (cclauss) [inline arithmetic_analysis/lu_decomposition.py:9]:
```suggestion
from numpy.typing import ArrayLike
```
And then on lines 14 and 15 change `NDArray` --> `ArrayLike`

**Comment by CaedenPH** [comment]
Please note that project euler problem 104 is failing, however I did not change this file.

**Comment by CaedenPH** [comment]
> Please note that project euler problem 104 is failing, however I did not change this file.

@cclauss what should I do about this?

**Review by cclauss — APPROVED** [review]
⭐ 🌟 ⭐ 
🚀 🥳 🎁

**Comment by cclauss** [comment]
If you are up for a smaller, easier one then please add [flake8-builtins](https://github.com/gforcada/flake8-builtins) to pre-commit.

**Comment by CaedenPH** [comment]
> If you are up for a smaller, easier one then please add [flake8-builtins](https://github.com/gforcada/flake8-builtins) to pre-commit.

Good morning, I will work on this when I have time

**Pull request merged**